### PR TITLE
Destination BigQuery + BigQuery denormalized: add configurable file buffer count

### LINKED ISSUE #20975: Destination BigQuery: configurable value for file buffer
## Tell us about the problem you're trying to solve
Introduces a configurable value for increasing the number of file buffers. The main premise for this functionality is that by increasing the number of file buffers for sources with interleaved data (e.g. Change Data Capture aka CDC) there will be an increase in performance due to a decrease in the number of buffer thrashing. This also will be important once parallel processing gets introduced since data will become interleaved at that point

[Reference PR](https://github.com/airbytehq/airbyte/pull/20879) and [PR that fixes bug](https://github.com/airbytehq/airbyte/pull/21890) which introduces the same configurability for Destination Redshift. Note the value has been selected with the understanding that currently Airbyte supports 1 GB of available memory for the destination connector. If this value changes then the range of the number of file buffers should also be adjusted

[Additional context](https://github.com/airbytehq/airbyte/pull/15707)

## Describe the solution you’d like
Introduce a configurable parameter within the spec, logic to retrieve the user-configured parameter, guard rails to prevent file buffers from exceeding a fixed limit, and tests to verify the number of file buffers falls within this range and does not drop below the previous default number of `10` file buffers

- [ ] BigQuery
- [x] Snowflake

## Describe the alternative you’ve considered or used
A clear and concise description of any alternative solutions or features you've considered or are using today.

## Are you willing to submit a PR?

Nope

### PR DESCRIPTION
**Pull request by cynthiaxyin**
## What

closes #20975

Allow users to configure the number of file buffers for destination BigQuery/BigQuery denormalized connectors. This improves performance for CDC sources with interleaved data where a low file buffer count would lead to buffer thrashing otherwise.

## How

Add new file_buffer_count parameter to the destination connector specs under the GCS staging loading method.

## Recommended reading order
1. Dockerfile
2. spec.json
3. BigQueryDestination.java
4. BigQuery*DestinationAcceptanceTest.java

## 🚨 User Impact 🚨

This is a minor version change that adds a new optional parameter to the specs.

Note that a file buffer has a default memory size of 31 MB, so OSS users should proceed with caution as the maximum configurable number of file buffers (50) may exceed memory expectations (1 GB).

## Pre-merge Checklist

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Connector version has been incremented
    - [ ] Version has been bumped according to our [Semantic Versioning for Connectors](https://docs.airbyte.com/contributing-to-airbyte/#semantic-versioning-for-connectors) guidelines
    - [ ] `Dockerfile` has updated version
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` with an entry for the new version. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)

- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</details>

### CODE REVIEW

File: airbyte-integrations/connectors/destination-bigquery-denormalized/Dockerfile [hunk 1]
```diff
@@ -17,5 +17,5 @@ ENV ENABLE_SENTRY true
 
 COPY --from=build /airbyte /airbyte
 
-LABEL io.airbyte.version=1.2.20
+LABEL io.airbyte.version=1.3.0
```
Comment (evantahler) [inline airbyte-integrations/connectors/destination-bigquery-denormalized/Dockerfile:20]: 👍 Following our [versioning guide](https://docs.airbyte.com/contributing-to-airbyte/#semantic-versioning-for-connectors), the addition of a new feature is a minor change

File: docs/integrations/destinations/bigquery-denormalized.md [hunk 2]
```diff
@@ -8,8 +8,8 @@ See [destinations/bigquery](https://docs.airbyte.com/integrations/destinations/b
 
 | Version | Date       | Pull Request                                               | Subject                                                                                                                  |
 | :------ | :--------- | :--------------------------------------------------------- | :----------------------------------------------------------------------------------------------------------------------- |
+| 1.3.0   | 2023-04-18 | [25287](https://github.com/airbytehq/airbyte/pull/25287)   | Add configurable parameter for number of file buffers                                                                                                      |
```
Comment (ryankfu) [inline docs/integrations/destinations/bigquery-denormalized.md:11]: Would add that this is isolated to using GCS as a loading method, both here and for regular `destination-bigquery`

File: docs/integrations/destinations/bigquery.md [hunk 3]
```diff
@@ -135,6 +135,7 @@ Now that you have set up the BigQuery destination connector, check out the follo
 
 | Version | Date       | Pull Request                                               | Subject                                                                                                                  |
 | :------ | :--------- | :--------------------------------------------------------- | :----------------------------------------------------------------------------------------------------------------------- |
+| 1.3.0   | 2023-04-18 | [25287](https://github.com/airbytehq/airbyte/pull/25287)   | Add configurable parameter for number of file buffers                                                                                                       |
```
Comment (ryankfu) [inline docs/integrations/destinations/bigquery.md:138]: Minor nit: maybe add `#` in front of the hypertext, seems it was broken from the last merge

**Review by ryankfu — APPROVED** [review]
LGTM, minor nit but not blocking

**Comment by ryankfu** [comment]
@johnwasserman Once this PR gets merged and published you should be able to increase the buffers for the connections using `destination-bigquery`

**Comment by cynthiaxyin** [comment]
/publish connector=connectors/destination-bigquery-denormalized

**Pull request merged**